src/features/device-state/routes/state.ts

```diff
@@ -25,6 +26,163 @@ export const varListInsert = (varList: EnvVarList, obj: Dictionary<string>) => {
+
+		const serviceInstall = (() => {
+			const si = serviceInstallFromImage(device, image);
+
+			if (si != null) {
+				return si;
+			}
+
+			if (image.is_a_build_of__service[0] == null) {
+				throw new Error(
+					`Could not find service install for device: '${
+						device.uuid
+					}', image: '${image?.id}', service: '${JSON.stringify(
+						image?.is_a_build_of__service,
+					)}', service installs: '${JSON.stringify(device.service_install)}'`,
+				);
+			}
+
+			/**
+			 * Return a fake service install as this is what we will use for MultiApp
+			 */
+			return {
+				service: [image.is_a_build_of__service[0]],
+				device_service_environment_variable: [],
+			};
+		})();
+
+		/**
+		 * Get the first service in the array, which will be tied to a service install
+		 */
+		const [service] = serviceInstall.service;
+
+		const environment: Dictionary<string> = {};
+		varListInsert(ipr.image_environment_variable, environment);
+		varListInsert(application.application_environment_variable, environment);
+		varListInsert(service.service_environment_variable, environment);
+		varListInsert(device.device_environment_variable, environment);
+		varListInsert(
+			serviceInstall.device_service_environment_variable,
+			environment,
+		);
+
+		const labels: Dictionary<string> = {};
+		[...ipr.image_label, ...service.service_label].forEach(
+			({ label_name, value }: { label_name: string; value: string }) => {
+				labels[label_name] = value;
+			},
+		);
+
+		_.each(ConfigurationVarsToLabels, (labelName, confName) => {
+			if (confName in config && !(labelName in labels)) {
+				labels[labelName] = config[confName];
+			}
+		});
+
+		const imgRegistry =
+			image.is_stored_at__image_location +
+			(image.content_hash != null ? `@${image.content_hash}` : '');
+
+		services[service.id] = {
+			imageId: image.id,
+			serviceName: service.service_name,
+			image: formatImageLocation(imgRegistry),
+			// This needs spoken about...
+			running: true,
+			environment,
+			labels,
+		};
+		// Don't send a null contract as this is a waste
+		// of bandwidth (a null contract is the same as
+		// the lack of a contract field)
+		if (image.contract != null) {
+			services[service.id].contract = image.contract;
+		}
+
+		if (
+			composition != null &&
+			composition.services != null &&
+			composition.services[service.service_name] != null
+		) {
+			const compositionService = composition.services[service.service_name];
+			// We remove the `build` properly explicitly as it's expected to be present
+			// for the builder, but makes no sense for the supervisor to support
+			delete compositionService.build;
+			services[service.id] = {
+				...compositionService,
+				...services[service.id],
+			};
+		}
+	});
+
+	return {
+		releaseId: release.id,
+		releaseVersion: release.release_version,
+		commit: release.commit,
+		name: application.app_name,
+		services,
+		networks: composition?.networks || {},
+		volumes: composition?.volumes || {},
```

Comment:
Should these be
```suggestion
		networks: composition?.networks ?? {},
		volumes: composition?.volumes ?? {},
```
?